src/plugins/vis_type_timeseries/server/validation_telemetry/validation_telemetry_service.ts

```diff
@@ -64,12 +64,14 @@ export class ValidationTelemetryService implements Plugin<ValidationTelemetrySer
         })
       );
     }
-    const internalRepository = core.savedObjects.createInternalRepository();
+    const internalRepository = core
+      .getStartServices()
+      .then(([start]) => start.savedObjects.createInternalRepository());
 
     return {
       logFailedValidation: async () => {
         try {
-          await internalRepository.incrementCounter(
+          await (await internalRepository).incrementCounter(
```

Comment:
So, I'm not sure this is the best implementation:

I wanted to avoid creating a new repository at every `logFailedValidation` call, so I create a chained promise for the repository outside of the function and then await it during the call.
Functionally this is what we want, not sure `await (await internalRepository).incrementCounter` is the cleaner way to implement it. Also, should `internalRepository` be named `internalRepositoryP`, or do we have a naming convention for promises?

WDYT?

Reply:
I'm fine with the current implementation, with a slight preference for
```ts
const internalRepositoryPromise = core.getStartServices()...

// ....

const internalRepository = await internalRepositoryPromise;
await internalRepository.incrementCounter(
```

for readability.